include/circt/Dialect/FIRRTL/FIRRTLCanonicalization.td

```diff
@@ -326,6 +336,80 @@ def XorOfPad : Pat <
                                  (XorPrimOp (TailPrimOp $x, (TypeWidthAdjust32 $x, $y)), $y))),
   [(KnownWidth $x), (UIntType $x), (EqualTypes $x, $pad)]>;
 
+/// orr(asSInt(x)) -> orr(x)
+def OrRasSInt : Pat <
+  (OrRPrimOp:$old (AsSIntPrimOp $x)),
+  (MoveNameHint $old, (OrRPrimOp $x)),
+  [(KnownWidth $x), (IntTypes $x)]>;
+
+/// orr(asUInt(x)) -> orr(x)
+def OrRasUInt : Pat <
+  (OrRPrimOp:$old (AsUIntPrimOp $x)),
+  (MoveNameHint $old, (OrRPrimOp $x)),
+  [(KnownWidth $x), (IntTypes $x)]>;
+
+/// orr(cvt(x)) -> orr(x)
```

Comment:
Please add tests for these, at least somewhat?

I'll go through these more carefully soon, but would be good to see them in action and ensure they're working as expected.